### PR TITLE
River network geometry

### PR DESCRIPTION
**Pull request by dblodgett-usgs**
As noted in #11 - I want to rework how to major rivers are selected. But this is ready for an initial merge.

### CODE REVIEW

**Comment by dblodgett-usgs** [comment]
Note that I have not tried to build what is contributed in 6_prep.

File: 1_fetch/src/fetch_river_geometry.R [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+
+sf_converter <- function(x)
+  sapply(x, function(x) try(sf::read_sf(x), silent = TRUE),
+         USE.NAMES = TRUE, simplify = FALSE)
+
+navigate_nldi <- function(f_id, f_source, mode = "UM",
+                          d_source = NULL, distance = NULL) {
+
+  url <- paste("https://cida.usgs.gov/nldi", f_source, f_id, "navigate", mode, d_source,
```
Comment (jread-usgs) [inline 1_fetch/src/fetch_river_geometry.R]: works the same, but I find `sprintf` to be a lot more readable for these kinds of things:
```r
url <- sprintf"https://cida.usgs.gov/nldi/%s/%s/navigate/%s/%s", f_source, f_id, mode, d_source)
```
not worth a change, but just a thought.

File: 6_vizprep/src/prep_rivers_fun.R [hunk 2]
```diff
@@ -0,0 +1,12 @@
+
+prep_rivers_fun <- function(ind_file, river_ind){
+
+  sf_rivers <- readRDS(sc_retrieve(river_ind))
+
+  plot_fun <- function(){
+    plot(sf_rivers$sf_gage_rivers$geometry, col = "deepskyblue", lwd = .5, add = TRUE)
```
Comment (jread-usgs) [inline 6_vizprep/src/prep_rivers_fun.R]: 👍

File: 1_fetch/src/fetch_river_geometry.R (hunk 1 above)
Comment (dblodgett-usgs) [reply 1_fetch/src/fetch_river_geometry.R]: yeah... I always think to myself, I should really be templating this, but never do.

**Pull request merged**